src/main/java/uk/gov/justice/hmpps/prison/service/CourtHearingReschedulingService.java

```diff
@@ -28,18 +27,23 @@
 public class CourtHearingReschedulingService {
 
     private final CourtEventRepository courtEventRepository;
+    private final OffenderBookingRepository offenderBookingRepository;
 
     private final Clock clock;
 
-    public CourtHearingReschedulingService(final CourtEventRepository courtEventRepository, final Clock clock) {
+    public CourtHearingReschedulingService(final CourtEventRepository courtEventRepository,
+                                           final OffenderBookingRepository offenderBookingRepository,
+                                           final Clock clock) {
         this.courtEventRepository = courtEventRepository;
+        this.offenderBookingRepository = offenderBookingRepository;
         this.clock = clock;
     }
 
     @Transactional
-    @VerifyBookingAccess(overrideRoles = "COURT_HEARING_MAINTAINER")
-    @PreAuthorize("hasRole('COURT_HEARING_MAINTAINER') and hasAuthority('SCOPE_write')")
     public CourtHearing reschedule(final Long bookingId, final Long hearingId, final LocalDateTime revisedDateTime) {
+        if (!offenderBookingRepository.existsById(bookingId)) {
+            throw EntityNotFoundException.withMessage("Offender booking with id %d not found.", bookingId);
+        }
```

Comment:
This check is needed in a few places because a side effect of @VerifyBookingAccess is that an exception is thrown if the booking is not found.